lib/client/db/profile/profile.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
```

Comment:
This is probably meant to land in `lib/client/db` instead - to be implemented by more specific packages in `lib/client/db/...`?

Reply:
Not really, this is intentional so particular implementations can reuse this struct from here. Can't move it level up due to circular imports.

Reply:
With the structs in `lib/client/db`, they'd do exactly that. The specific provider implementation would use the struct from `lib/client/db`. Of course this means `lib/client/db/<provider>` cannot be used from `lib/client/db` which does cause a circular dependency. But it should not have to - the providers should just register themselves somewhere in the root and the root would use them via an interface - does that make sense?

Reply:
Yeah, I know what you're saying, I just feel that introducing this sort of registry when I only need to share one struct between a couple of packages may be a bit of an overkill.